src/extension-common/src/lib.rs

```diff
@@ -30,6 +30,28 @@ macro_rules! t_print_cli_err {
     };
 }
 
+#[derive(Debug)]
+pub struct FluvioExtensionMetaData {
+    pub command: String,
+    pub description: String,
+    pub version: String,
+}
+
+use structopt::StructOpt;
+pub trait FluvioExtension: StructOpt {
+    fn version(&self) -> String;
```

Comment:
use `&str`

Reply:
I don't think there's any way to implement this using `&str` if you want to be able to use `env!()` values.

Reply:
`env!` returns a `&str` but if we use the default implementation for `metadata`, it's just converting those to `String` which seems not ideal to me.

Reply:
The biggest issue I have with `env!` is for the `CARGO_PKG_DESCRIPTION` environment variable, is set at the crate level and the current `extension-consumer` crate has `topic`, `partition`, `produce`, `consume` which results in them having the same `description` in the metadata.